File: python/n66_AddOne.py
# https://leetcode-cn.com/problems/plus-one/
class Solution:
    def plusOne(self, digits):
        if digits[-1] != 9:
            digits[-1] += 1
        elif len(digits) > 1:
            digits[-1] = 0
            digits = self.plusOne(digits[:-1]) + digits[-1:]
        else:
            digits[-1] = 0
            digits.insert(0, 1)
        return digits
            
if __name__ == "__main__":
    solu = Solution()
    digits = [9, 9, 9, 9]
    print(solu.plusOne(digits))

File: cpp/n66_AddOne.cpp
// https://leetcode-cn.com/problems/plus-one/
#include<iostream>
#include<vector>
using namespace std;
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        for(int i = digits.size()-1; i >= 0; --i){
            if(digits[i] == 9)digits[i] = 0;
            else{
                digits[i]++;
                return digits;
            }
        }
        digits.push_back(0);
        digits[0] = 1;
        return digits;
    }
};

int main(){
    vector<int> digits = {4, 3, 2, 3};
    Solution solu;
    auto result = solu.plusOne(digits);
    for(auto iter = result.begin(); iter != result.end(); ++iter) cout << *iter << ", ";
    cout << endl;
}